components/freertos/tasks.c

```diff
@@ -4449,7 +4450,7 @@ For ESP32 FreeRTOS, vTaskExitCritical implements both portEXIT_CRITICAL and port
 				pcWriteBuffer = prvWriteNameToBuffer( pcWriteBuffer, pxTaskStatusArray[ x ].pcTaskName );
 
 				/* Write the rest of the string. */
-				sprintf( pcWriteBuffer, "\t%c\t%u\t%u\t%u\r\n", cStatus, ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority, ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark, ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber );
+				sprintf( pcWriteBuffer, "\t%c\t%u\t%u\t%u\t%hd\r\n", cStatus, ( unsigned int ) pxTaskStatusArray[ x ].uxCurrentPriority, ( unsigned int ) pxTaskStatusArray[ x ].usStackHighWaterMark, ( unsigned int ) pxTaskStatusArray[ x ].xTaskNumber, ( int ) pxTaskStatusArray[ x ].xCoreID );
```

Comment:
Some existing code may rely on the format of the string returned by `vTaskList`. Suggest introducing a new sdkconfig option, e.g. CONFIG_FREERTOS_VTASKLIST_INCLUDE_COREID, disabled by default. Then put the modifications under `ifdef`, so that by default the old behavior is preserved.

Reply:
And here int could be "unsigned int" or "uint32_t" following above comment.

Reply:
Added the option CONFIG_FREERTOS_VTASKLIST_INCLUDE_COREID as requested. Please se the revised Pull Request.